File: FDK/SoftFX.Extended.Net/Exceptions/LogoutException.cs
namespace SoftFX.Extended.Errors
{
    using System;
    using System.Runtime.Serialization;
    using System.Security.Permissions;
    using SoftFX.Extended.Core;

    /// <summary>
    /// This exception indicates that a synchronous call has been interrupted by logout event.
    /// </summary>
    [Serializable]
    public class LogoutException : RuntimeException
    {
        internal LogoutException(string message)
            : base(HResults.FX_CODE_ERROR_LOGOUT, message)
        {
        }

        /// <summary>
        /// 
        /// </summary>
        /// <param name="info"></param>
        /// <param name="context"></param>
        [SecurityPermissionAttribute(SecurityAction.Demand, SerializationFormatter = true)]
        protected LogoutException(SerializationInfo info, StreamingContext context)
            : base(info, context)
        {
        }
    }
}

File: FDK/FDK/Core/LogoutException.cpp
#include "stdafx.h"
#include "LogoutException.h"


CLogoutException::CLogoutException()
    : std::runtime_error("Client has been logged out.")
{
}

CLogoutException::CLogoutException(const char* message)
    : std::runtime_error(message)
{
}

CLogoutException::CLogoutException(const std::string& message)
    : std::runtime_error(message.c_str())
{
}
